### PR TITLE
Added the Recommendations feature documentation

### PR DESCRIPTION
**Pull request by 1akshitha**
Recommendation feature Documentation. Related git issue :https://github.com/wso2/product-apim/issues/7227

### CODE REVIEW

File: en/docs/wip/api-recommendations.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+- **max_recommendations**: Max recommendations displayed in the developer portal. (By default 6 APIs are displayed in
+ the developer portal)
+- **wait_duration**: Recommendations for users are requested from the API in 15 minutes intervals. If that interval
+ needs to be changed, add the time in minutes.
+- **apply_for_all_tenants**: In a multi tenant deployment, if recommendations feature is enabled, it will show
+ recommendations in all the developer portals and if you need to apply the recommendation feature only for selected 
+ tenants, add false here. 
+
+### Enable recommendations feature for a specific tenant
```
Comment (fazlan-nazeem) [inline en/docs/wip/api-recommendations.md]: shall we reword it to **only for specific tenants**? I assume it will be enabled to all tenants if it is enabled in the deployment.toml.

File: en/docs/wip/api-recommendations.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ - Recommending APIs based on the application developers interests and the application that they are creating.
+
+Using AI technologies, this feature will analyze the behavior of the developer, application that he or she is 
+developing and suggest a list of new APIs that are not subscribed by the user but might be useful for the project. 
+In order to recommend effectively, there must be at least 20 APIs in the portal.
+
+![](../assets/img/api-recommendations.png)
+
+## Solution
+Recommendation server will be deployed in WSO2 and will be provided to all the customers as a service. This service
```
Comment (fazlan-nazeem) [inline en/docs/wip/api-recommendations.md]: Recommendation service is available as a SaaS service for all wso2 customers and secured via oauth2. If a specific customer requires to enable this feature, they will have to request a consumer key and secret via the wso2 support portal.

File: en/docs/wip/api-recommendations.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+developing and suggest a list of new APIs that are not subscribed by the user but might be useful for the project. 
+In order to recommend effectively, there must be at least 20 APIs in the portal.
+
+![](../assets/img/api-recommendations.png)
+
+## Solution
+Recommendation server will be deployed in WSO2 and will be provided to all the customers as a service. This service 
+is OAuth2 protected and if you want to use this feature in your developer portal, APIM team will provide access to 
+the recommendations API. Once the recommendations feature is activated, recommendation feature related events will be 
+published to this recommendations API. When a user logs in to the developer portal, recommended APIs based on that
```
Comment (fazlan-nazeem) [inline en/docs/wip/api-recommendations.md]: The recommendation service analyzes all the information and runs through its algorithm to find the most suitable API recommendations for the logged-in user. These APIs will then be listed in the devportal similar to the following image?

File: en/docs/wip/api-recommendations.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
+
+# AI based Recommendation System for the API Developer portal
```
Comment (fazlan-nazeem) [inline en/docs/wip/api-recommendations.md]: API Deveoper Portal -> Developer Portal

File: en/docs/wip/api-recommendations.md (hunk 1 above)
Comment (1akshitha) [reply en/docs/wip/api-recommendations.md]: Fixed

File: en/docs/wip/api-recommendations.md (hunk 2 above)
Comment (1akshitha) [reply en/docs/wip/api-recommendations.md]: Fixed

File: en/docs/wip/api-recommendations.md (hunk 3 above)
Comment (1akshitha) [reply en/docs/wip/api-recommendations.md]: Fixed

File: en/docs/wip/api-recommendations.md (hunk 4 above)
Comment (1akshitha) [reply en/docs/wip/api-recommendations.md]: Fixed
